File: Python3/Medium/WrongAnswer/most-popular-video-creator_1.py
class Solution:
    def mostPopularCreator(self, creators: List[str], ids: List[str], views: List[int]) -> List[List[str]]:
        
        d = {}
        for i in range(len(creators)):
            if creators[i] not in d:
                d[creators[i]] = [(ids[i], views[i])]
            else:
                d[creators[i]].append((ids[i], views[i]))
        # print(d)
        max_views = 0
        res = []
        for k, v in d.items():
            views = 0
            for vid in v:
                views += vid[1]
            if views > max_views:
                max_views = views
                res = []
                res.append(k)
            elif views == max_views:
                res.append(k)
        # print(res)
        ans = []
        for r in res:
            ans.append([r, max(d[r], key=lambda x: x[1])[0]])
        return ans

File: Java/Medium/WrongAnswer/most-popular-video-creator_1.java
class Solution {
    public List<List<String>> mostPopularCreator(String[] creators, String[] ids, int[] views) {
        
        Map<String, Map<String, Integer>> map = new HashMap<>();
        for (int i = 0; i < creators.length; i++) {
            if (!map.containsKey(creators[i])) {
                map.put(creators[i], new HashMap<>());
            }
            Map<String, Integer> temp = map.get(creators[i]);
            temp.put(ids[i], temp.getOrDefault(ids[i], 0) + views[i]);
        }
        String max = "";
        int maxCount = 0;
        for (String creator : map.keySet()) {
            int count = 0;
            for (String id : map.get(creator).keySet()) {
                count += map.get(creator).get(id);
            }
            if (count > maxCount || (count == maxCount && creator.compareTo(max) < 0)) {
                max = creator;
                maxCount = count;
            }
        }
        List<List<String>> res = new ArrayList<>();
        List<String> temp = new ArrayList<>();
        temp.add(max);
        temp.add(findMax(map.get(max)));
        res.add(temp);
        return res;
    }
    
    private String findMax(Map<String, Integer> map) {
        String max = "";
        int maxCount = 0;
        for (String id : map.keySet()) {
            if (map.get(id) > maxCount || (map.get(id) == maxCount && id.compareTo(max) < 0)) {
                max = id;
                maxCount = map.get(id);
            }
        }
        return max;
    }
}